frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/GitMacheteGraphTableManager.java

```diff
@@ -203,8 +207,14 @@ private void updateRepository(Path mainDirectoryPath, Path gitDirectoryPath, boo
       // (or something else happens)
       // For state that caused unexpected behavior see tag `strange-vavr-try-behavior`
       try {
-        IBranchLayout branchLayout = updateBranchLayoutAndMacheteFilePath();
-        gitMacheteRepositoryRef.set(gitMacheteRepositoryFactory.create(mainDirectoryPath, gitDirectoryPath, branchLayout));
+        GitRepository gitRepository = gitRepositorySelectionProvider.getSelectedRepository();
+        if (gitRepository != null) {
+          Path macheteFilePath = getMacheteFilePath(gitRepository);
+          IBranchLayout branchLayout = createBranchLayout(macheteFilePath);
+          graphTable.setBranchLayout(branchLayout);
+          graphTable.setMacheteFilePath(macheteFilePath);
+          gitMacheteRepositoryRef.set(gitMacheteRepositoryFactory.create(mainDirectoryPath, gitDirectoryPath, branchLayout));
+        }
```

Comment:
Maybe `else` with some logging (and maybe even message for user) in case when something unexpected (like `gitRepository == null`) will happen?

Reply:
Done, added just a `LOG.warn` (this shouldn't be displayed to do user, right?)

Reply:
You are right. It's only displayed in logs. In general all logging methods are not directly visible for user.